crates/zksync/core/src/vm/runner.rs

```diff
@@ -259,7 +263,13 @@ where
         error!("balance is 0 for {caller:?}, transaction will fail");
     }
     let max_fee_per_gas = fix_l2_gas_price(ecx.env.tx.gas_price.to_u256());
-    let gas_limit = fix_l2_gas_limit(ecx.env.tx.gas_limit.into(), max_fee_per_gas, value, balance);
+
+    let gas_limit = if paymaster_params.paymaster != Address::ZERO.to_h160() {
+        // If paymaster is set, use the proposed gas limit
+        ecx.env.tx.gas_limit.into()
```

Comment:
I'm curious why we don't need to apply the fix when using paymaster. If anything the paymaster gaslimit must adhere to the patch we apply

Reply:
In the case of the paymaster we want to avoid the fix_l2_gas_limit as it takes into account the user balance and in the case users have very little the gas limit it proposes does not make it through the validation of the bootloader as its two low to finish the frame execution.